starport/cmd/network_chain_publish.go

```diff
@@ -0,0 +1,151 @@
+package starportcmd
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/manifoldco/promptui"
+	"github.com/spf13/cobra"
+	"github.com/tendermint/starport/starport/pkg/clispinner"
+	"github.com/tendermint/starport/starport/pkg/cosmosaccount"
+	"github.com/tendermint/starport/starport/pkg/events"
+	"github.com/tendermint/starport/starport/pkg/xurl"
+	"github.com/tendermint/starport/starport/services/network"
+)
+
+const (
+	flagTag     = "tag"
+	flagBranch  = "branch"
+	flagHash    = "hash"
+	flagGenesis = "genesis"
+	flagNoCheck = "no-check"
+)
+
+// NewNetworkChainPublish returns a new command to publish a new chain to start a new network.
+func NewNetworkChainPublish() *cobra.Command {
+	c := &cobra.Command{
+		Use:   "publish [source-url]",
+		Short: "Publush a new chain to start a new network",
+		Args:  cobra.ExactArgs(1),
+		RunE:  networkChainPublishHandler,
+	}
+
+	c.Flags().String(flagBranch, "", "Git branch to use for the repo")
+	c.Flags().String(flagTag, "", "Git tag to use for the repo")
+	c.Flags().String(flagHash, "", "Git hash to use for the repo")
+	c.Flags().String(flagGenesis, "", "URL to a custom Genesis")
+	c.Flags().Bool(flagNoCheck, false, "Skip verifying chain's integrity")
+	c.Flags().String(flagFrom, cosmosaccount.DefaultAccount, "Account name to use for sending transactions to SPN")
+	c.Flags().AddFlagSet(flagSetKeyringBackend())
+	c.Flags().AddFlagSet(flagSetHome())
+	c.Flags().AddFlagSet(flagSetYes())
+
+	return c
+}
+
+func networkChainPublishHandler(cmd *cobra.Command, args []string) error {
+	var (
+		source        = args[0]
+		tag, _        = cmd.Flags().GetString(flagTag)
+		branch, _     = cmd.Flags().GetString(flagBranch)
+		hash, _       = cmd.Flags().GetString(flagHash)
+		genesisURL, _ = cmd.Flags().GetString(flagGenesis)
+		noCheck, _    = cmd.Flags().GetBool(flagNoCheck)
+	)
+
+	s := clispinner.New()
+	defer s.Stop()
+
+	ev := events.NewBus()
+	go printEvents(ev, s)
```

Comment:
without a graceful shutdown, this goroutine can die without printing the events

Reply:
1b18c98